### PR TITLE
Milestone 4 filtering

### PR DESCRIPTION
**Pull request by ikraamg**
[Live Link](https://bookstore-ip.herokuapp.com/)

In this milestone, we completed the following tasks:

Enhanced the Redux store by adding a filter property
Added a CategoryFilter component that renders a category selection
Implement HandleFilterChange method and logic for filtering the bookList

Please find our commit below,
Thank you.

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const filterReducer = (state = '', action) => {
```
Comment (mosaaleb) [inline src/reducers/filter.js]: - [x] state should default to `all`

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done @ikraamg @Phylis05 :clap:

All requirements are handled, only one left. 

Please change the filter reducer default state value, and submit a new code review.

##### Muhammad Ebeid.

**Comment by ikraamg** [comment]
Hi @mosaaleb , i was just confused as to how should we keep the default of the filter reducer method to 'all' but also not have 'all' stored in the redux store. Is that not a contradiction in the requirements?

**Review by jofer86 — APPROVED** [review]
## `const PROJECT_STATUS = "APPROVED"` 🚀 🚀 🚀 
## :wave: :wave: Hello! I'm Jorge, and  I will be conducting this code review. Let me get a closer look 🔎 🔍 🔎 🔍 🕵🏼 
- All of the required changes are taken care off great job you may move along.
--------------------------------------------------------------------
**Rock on**  🤘🏼 
**_Jorge Fernando Rincón Figueroa._** 
Technical Support Engineer.

**Pull request merged**